File: app/repository/Repository.scala
package repository

import domain.{Team, Person, Query}

trait Repository {
    def findTeamForPerson(person: Person): Team
    def listPersons: java.util.List[Person]
    def listTeams: List[Team]
    def findPersonsForTeam(team: Team): List[Person]
    def queryPersons(query: Query): java.util.List[Person]
    def queryTeams(query: Query): java.util.List[Team]
}

File: app/data/Repository.java
package data;

import domain.Person;
import domain.Query;
import domain.Team;

import java.util.List;

public interface Repository {
    List<Person> listPersons();
    List<Team> listTeams();

    List<Person> queryPersons(Query query);
    List<Team> queryTeams(Query query);

    void addPerson(Person person);
}
